### DISCUSSION TITLE
Authentication using Token System

### DISCUSSION THREAD

**Discussion by agarwalrohit2503** [serverpod/serverpod#281] [Q&A] [answered]
Is there any tokenisation system developed for apis
like how to authenticate that the api request is from a valid source or not?

**Comment by vlidholt** [comment, marked as answer]
Serverpod comes with a module for handling authentication (or you can wing it yourself if you prefer).

This should get you started:
https://docs.serverpod.dev/concepts/authentication

To check if a user is signed in you can check the `session.auth.authenticatedUserId` property of the `Session` object. If it is non-null, the user is signed in.

You can also assign scopes to your `Endpoint` to restrict access to all methods of an endpoint. Override the `requiredScopes` method. E.g.

```dart
@override
Set<Scope> get requiredScopes => {Scope.admin};
```

Or if you just want to restrict access to signed in users, you can override the `requireLogin` method.

```dart
@override
bool get requireLogin => false;
```

**Comment by agarwalrohit2503** [comment]
What if i want to expire the loggedIn session after sometime without user intervention

**Reply by vlidholt** [reply]
You can use a Future call to do that, with the `session.auth.signOut` method.

**Reply by agarwalrohit2503** [reply]
Ok thank you so much
